### PR TITLE
Lesson 3 Pull Request

### PR DESCRIPTION
**Pull request by IanMayther**
Request serves as submitter for Exercise 3.1, 3.2, 3.3 and Mailroom  Part 1

### CODE REVIEW

File: students/IanMayther/Lesson03/list_lab.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    #Third Bullet
+    response = input("Can you add another fruit?")
+    fruits.append(response)
+
+    #Fourth Bullet
+    print(fruits)
+
+    #Fifth Bullet
+    number = input(f"Please provide a number between 1 and {len(fruits)}?")
```
Comment (charlesrobison) [inline students/IanMayther/Lesson03/list_lab.py:21]: Nice!

File: students/IanMayther/Lesson03/list_lab.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    if int(number) > len(fruits):
+        print(f"Value too large, using {len(fruits)}")
```
Comment (charlesrobison) [inline students/IanMayther/Lesson03/list_lab.py:23]: Very good.

File: students/IanMayther/Lesson03/list_lab.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    #Seventh Bullet
+    response = input("Can you add another fruit?")
+    fruits.insert(0, response)
+    print(fruits)
+
+    #Eighth Bullet
+    p_fruits = []
+    for item in fruits:
+        if item[:1] == "P" or item[:1] == "p":
```
Comment (charlesrobison) [inline students/IanMayther/Lesson03/list_lab.py:41]: Good - you can use upper() or the like as well.

File: students/IanMayther/Lesson03/list_lab.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        while definite == False:
+            response = input(f"Do you like {fruits[i].lower()}? Y/N - ")
+            if response == "Y":
+                definite = True
+                fruits3.append(fruits[i])                
+            elif response == 'N':
+                definite = True
+            else:
+                print("Please select Y or N") 
+                definite = False
```
Comment (charlesrobison) [inline students/IanMayther/Lesson03/list_lab.py:83]: Nicely done

File: students/IanMayther/Lesson03/list_lab.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    #Second Bullet
+    print("Removing Last Item.")
+    fruits.pop()
+    
+    #Third Bullet
+    print(f"New List: {fruits}")
+
+    #Fourth Bullet
+    fruits2 = fruits[:]
+    Not_In_List = False
```
Comment (charlesrobison) [inline students/IanMayther/Lesson03/list_lab.py:58]: This variable should be snake_case too, just like all your others.

File: students/IanMayther/Lesson03/strformat_lab.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    elif inputs[0] >= 10:
+        file_num = f"file_0{inputs[0]}"
+    elif inputs[0] > 99:
+        file_num = f"file_{inputs[0]}"  
+    deliv1 = file_num
+    
+    #Display (2) decimal places
+    flo_num = str(inputs[1])
+    output = ""
+    Need_Decimal = False
```
Comment (charlesrobison) [inline students/IanMayther/Lesson03/strformat_lab.py:30]: This should be snake_case

File: students/IanMayther/Lesson03/strformat_lab.py [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    names = ['Ian', 'Jill', 'Spencer', 'Ryan', 'Dan']
+    ages = [12, 36, 57, 34, 39]
+    cost = [10.00, 1000.00, 0.00, 25.00, 6.00]
+
+    print("{0:>15s}{1:^15s}{2:>12s}".format("Name", "Age", "Cost"))
+
+    for name in range(len(names)):
+        print("{0:>15s}".format(names[name]), end=' ')
+        print("{0:^15}".format(ages[name]), end=' ')
+        print("{0:>10.2f}".format(cost[name]))
```
Comment (charlesrobison) [inline students/IanMayther/Lesson03/strformat_lab.py:101]: Looks good

File: students/IanMayther/mailroom/Mailroom.py [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        try:
+            value = float(input("What is the value of the donation: "))
+            break      
+        except ValueError:
+            print("Not a valid donation value")
+    return value
+
+
+#Print Email
+def email(TO, gift_amount):
```
Comment (charlesrobison) [inline students/IanMayther/mailroom/Mailroom.py:54]: TO should be lowercase - might use something a bit more descriptive too.

File: students/IanMayther/mailroom/Mailroom.py [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+        col = my_List[i][1]
+        #Sum Total 
+        total = sum(col)
+        #Gift Count  
+        No_Gifts = len(my_List[i][1])
+        #Calc Average
+        Ave_Gift = total / No_Gifts
+        #Print Table
+        print("{0:<25s}${1:>14.2f}{2:>17d}  ${3:>11.2f}".format(my_List[i][0],total, No_Gifts, Ave_Gift, end =''))    
+    return
```
Comment (charlesrobison) [inline students/IanMayther/mailroom/Mailroom.py:87]: Missing the sorting requirement - donors should be sorted by total given, in descending order

File: students/IanMayther/mailroom/Mailroom.py [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+            elif directive == "2":
+                #Launch Create Report
+                print("Create Report")
+                print_report(donors1)
+            elif directive == "3":
+                #Quit
+                print("Quitter")
+                real_response = True
+        else:
+            print("Please select one of the options above")
```
Comment (charlesrobison) [inline students/IanMayther/mailroom/Mailroom.py:129]: Please make it possible to add a new donor after entering 'list'

**Comment by IanMayther** [comment]
Good evening Charles,

Can you please elaborate on the snake_case?  I am not sure I understand.

Regarding your comments on sorting the vendors, it now explains why I
thought that portion of the assignment seemed easy and straightforward.

Thanks,
Ian

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by natasha-aleksandrova** [comment]
@IanMayther snake_case is a style used to name variable and functions in python

for example:


snake_case: `need_decimal` all lower case words separated by underscore
camelCase: `needDecimal` first word is lower case the rest is capitalized.

**Pull request merged**